hibernate-core/src/main/java/org/hibernate/persister/collection/OneToManyPersister.java

```diff
@@ -390,7 +391,8 @@ public RestrictedTableMutation<JdbcMutationOperation> generateDeleteAllAst(Mutat
 				keyRestrictionBindings,
 				null,
 				parameterBinders,
-				sqlWhereString
+				sqlWhereString,
+				Expectations.NONE
```

Comment:
Shouldn't the fix rather be in `org.hibernate.persister.collection.AbstractCollectionPersister#buildCollectionTableMapping` where we create the `MutationDetails` with a certain `Expectation` for the delete all part?

Reply:
We would have to change not the *delete all* part `MutationDetails` but the update one, since the `org.hibernate.persister.collection.OneToManyPersister#generateDeleteAllAst` method creates an update operation for the one-shot delete.

Since we have a `@JoinColumn`, the related table is the child entity one and not a dedicated relationship table. It would work, but then all updates on the related entity's table would have no Expectation, so I though creating a dedicated case just for the one-shot update would be better.

Reply:
Hmm, maybe @sebersole has some thoughts on that.

Reply:
`MutationDetails` and command creation get a bit odd when it comes to collections because of this "some deletes are updates" situation.  To be honest, I'm surprised this has not caused more problems.

I think the solution here is fine.